proposed/apphost-embed-install-location.md

```diff
@@ -0,0 +1,158 @@
+
+We can allow selection of which search locations the `apphost` will use. When
+search locations are configured, only the specified locations will be searched.
+
+```c
+enum search_location
+{
+    sl_default = 0,
+    sl_app_local = 1 << 0,
+    sl_embedded = 1 << 1,
+    sl_environment_variables = 1 << 2,
+    sl_global = 1 << 3,
+};
+```
+
+The behaviour corresponding to the `default` value is defined by the `apphost`.
+It is currently `app_local | environment_variables | global`.
+
+The search location could be specified via a property in the project:
+
+```xml
+<AppHostDotNetSearch>Global</AppHostDotNetSearch>
+```
+
+where the valid values are `AppLocal`, `Embedded`, `EnvironmentVariables`, or
+`Global`. Multiple values can be specified, delimited by semi-colons.
+
+When a value is specified, only those locations will be used. For example, if
+`Global` is specified, `apphost` will only look at global install locations, not
+app-local, at any embedded path, or at environment variables.
+
+### Embedded relative path to install location
+
+When a relative path is embedded and the `apphost` is [configured to look at it](#configuration-of-search-locations),
+that path will be used as the .NET install root when running the application.
+
+The install location could be specified via a property in the project:
+
+```xml
+<AppHostDotNetRoot>./path/to/runtime</AppHostDotNetRoot>,
+```
+
+Setting this implies `AppHostDotNetSearch=Embedded`. If `AppHostDotNetSearch` is
+explicitly set to a value that does not include `Embedded`, `AppHostDotNetRoot`
+is meaningless - the SDK will not write the relative path into the `apphost`
+and the `apphost` will not check for an embedded relative path.
+
+## Updated behaviour
+
+At a high level, the updated process and priority for `apphost` determining the
+install location would be:
+
+  1. App-local, if search location not configured
+      - Look for the runtime in the app's folder (self-contained apps)
+  2. Embedded, if specified as a search location
+      - Use the path embedded into `apphost`, relative to the app location
+  3. Environment variables, if search location not configured or if set as a
+  search location
+      - Read the `DOTNET_ROOT_<arch>` and `DOTNET_ROOT` environment variables
+  4. Global install (registered), if search location not configured or if set as
+  a search location
+      - Check for a registered install - registry key on Windows or a install
+      location file on non-Windows
+  5. Global install(default), if search location not configured or if set as a
+  search location
+      - Fall back to a well-known default install location based on the platform
+
+Be default - that is, without any embedded install location options - the
+effective behaviour remains as in the [current state](#state-in-net-8).
+
+## Considerations
+
+### Writing embedded options on build
+
+This proposal writes the install location options in the `apphost` on `build`.
```

Comment:
I think it is important that F5 continues to work even with apps with configured search locations. The easiest way to make that happen is by patching the apphost with the search locations for publish only.

Reply:
Absolutely agree - I suspect that if we do it as proposed, we'll immediately get lot of questions why VS/VS Code doesn't work at all. Another gesture which would break is `dotnet run`. And arguably it would be "our job" to fix these. In any case, eventually when we do have an SDK experience for this, all of the above should "just work" and so either this or SDK would have compensate for it anyway.